dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreDbClient.cs

```diff
@@ -0,0 +1,260 @@
+            using NpgsqlCommand cmd = commandInfo.ToNpgsqlCommand(connection);
+            await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task CreateVectorIndexAsync(string tableName, VectorStoreRecordVectorProperty vectorProperty, CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrEmpty(vectorProperty.IndexKind))
+        {
+            return;
+        }
+
+        NpgsqlConnection connection = await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+        await using (connection)
+        {
+            var commandInfo = this._sqlBuilder.BuildCreateVectorIndexCommand(this._schema, tableName, vectorProperty);
+            using NpgsqlCommand cmd = commandInfo.ToNpgsqlCommand(connection);
+            await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task DeleteTableAsync(string tableName, CancellationToken cancellationToken = default)
+    {
+        NpgsqlConnection connection = await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+        await using (connection)
+        {
+            var commandInfo = this._sqlBuilder.BuildDropTableCommand(this._schema, tableName);
+            using NpgsqlCommand cmd = commandInfo.ToNpgsqlCommand(connection);
+            await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task UpsertAsync(string tableName, Dictionary<string, object?> row, string keyColumn, CancellationToken cancellationToken = default)
+    {
+        NpgsqlConnection connection = await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+        await using (connection)
+        {
+            var commandInfo = this._sqlBuilder.BuildUpsertCommand(this._schema, tableName, keyColumn, row);
+            using NpgsqlCommand cmd = commandInfo.ToNpgsqlCommand(connection);
+            await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task UpsertBatchAsync(string tableName, IEnumerable<Dictionary<string, object?>> rows, string keyColumn, CancellationToken cancellationToken = default)
+    {
+        NpgsqlConnection connection = await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+        await using (connection)
+        {
+            var commandInfo = this._sqlBuilder.BuildUpsertBatchCommand(this._schema, tableName, keyColumn, rows.ToList());
+            using NpgsqlCommand cmd = commandInfo.ToNpgsqlCommand(connection);
+            await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+        }
```

Comment:
Many of these methods are the same exception for the command info.  Consider creating a method that takes teh commandInfo as a parameter and does the connection open and command conversion and execution.